Tests/Marketplace/search_and_install_packs.py

```diff
@@ -35,47 +34,49 @@
 WLM_TASK_FAILED_ERROR_CODE = 101704
 
 
-def is_pack_deprecated(pack_path: str) -> bool:
-    """Checks whether the pack is deprecated.
-    Tests are not being collected for deprecated packs and the pack is not installed in the build process.
+def is_pack_deprecated(pack_id: str, check_locally: bool = True, pack_api_data: dict | None = None) -> bool:
+    """
+    Check whether a pack is deprecated or not.
+    Can be checked locally (pack_metadata.json), or using Marketplace API response data.
+
     Args:
-        pack_path (str): The pack path
+        pack_id (str): ID of the pack to check.
+        check_locally (bool): Whether to check locally (pack_metadata file) or not (will use Marketplace API data instead).
```

Comment:
why do we need to check locally? What is the use case to check locally instead query the api?

Reply:
@ilaner When we're in post-update, the new bucket is created using the branch and not 'master'.
This means, that if I, for example, deprecate a pack named 'X' in my PR, this pack won't be uploaded to the bucket.

Because of this, if I send a marketplace API call to fetch information about 'X' (to check if it's deprecated) during the post-update phase (where the bucket is updated to the branch), the server returns a '400' response code, saying that this ID doesn't exist (since it's not on the marketplace).

We could perhaps just catch such exceptions, and deduce that if we get a '400' status code, the pack is deprecated.
Not sure if this is a good idea, though, and if there can be other cases where we get a '400' response code and want it to fail the build.